From 0a110b4b5182c2bcb7ae9d7a1c5ce4f4a6744c7c Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Tue, 7 Apr 2020 14:17:57 -0700
Subject: [PATCH] CMakeLists.txt: fix build without catkin testing enabled

* use CATKIN_ENABLE_TESTING conditional as suggested in:
  http://wiki.ros.org/roslaunch#Catkin
* it's not really clear that roslaunch_add_file_check shouldn't
  be called at all or roslaunch_add_file_check should check CATKIN_ENABLE_TESTING
  before trying to call catkin_run_tests_target inside
* see:
  https://answers.ros.org/question/200359/purpose-of-roslaunch_add_file_check/
  similar fix for imu_transformer in gentoo:
  https://github.com/gentoo/gentoo/commit/452c8600bbe989ad3b4f06211cdbdccfe5eb2f52

Upstream-Status: Pending

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5737763..ae11b7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,9 @@ find_package(catkin REQUIRED COMPONENTS roslaunch)
 
 catkin_package()
 
-roslaunch_add_file_check(launch/cartographer_demo.launch)
+if (CATKIN_ENABLE_TESTING)
+  roslaunch_add_file_check(launch/cartographer_demo.launch)
+endif()
 
 install(DIRECTORY launch config
   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
